packages/frontend/core/src/components/workspace-selector/workspace-card/quota-check.tsx

```diff
@@ -0,0 +1,188 @@
+import { useConfirmModal } from '@affine/component';
+import { GlobalDialogService } from '@affine/core/modules/dialogs';
+import { WorkspaceQuotaService } from '@affine/core/modules/quota';
+import { type I18nString, useI18n } from '@affine/i18n';
+import {
+  useLiveData,
+  useService,
+  useServiceOptional,
+  type WorkspaceMetadata,
+} from '@toeverything/infra';
+import { useCallback, useEffect } from 'react';
+
+import { useWorkspaceInfo } from '../../hooks/use-workspace-info';
+import * as styles from './styles.css';
+
+interface Message {
+  title: I18nString;
+  description: I18nString;
+  confirmText: I18nString;
+  tips?: I18nString[];
+  cancelText?: I18nString;
+}
+
+/**
+ *
+ * Notification that the cloud workspace quota has exceeded the limit
+ *
+ */
+export const QuotaCheck = ({
+  workspaceMeta,
+}: {
+  workspaceMeta: WorkspaceMetadata;
+}) => {
+  const { openConfirmModal } = useConfirmModal();
+  const workspaceQuota = useServiceOptional(WorkspaceQuotaService)?.quota;
+  const workspaceInfo = useWorkspaceInfo(workspaceMeta);
+  const quota = useLiveData(workspaceQuota?.quota$);
+  const usedPercent = useLiveData(workspaceQuota?.percent$);
+  const isOwner = workspaceInfo?.isOwner;
+  const globalDialogService = useService(GlobalDialogService);
+  const t = useI18n();
+
+  const onConfirm = useCallback(() => {
+    if (!isOwner) {
+      return;
+    }
+    globalDialogService.open('setting', {
+      activeTab: 'plans',
+      scrollAnchor: 'cloudPricingPlan',
+    });
+  }, [globalDialogService, isOwner]);
+
+  useEffect(() => {
+    if (workspaceMeta.flavour === 'local' || !quota) {
+      return;
+    }
+    const memberOverflow = quota.memberCount > quota.memberLimit;
```

Comment:
do we need revalidate WorkspaceQuota ?

Reply:
This dialog will only appear once when entering the workspace, and should not need to be revalidated？

Reply:
Yes, revalidate is required, otherwise quote data will not be loaded

Reply:
```
useEffect(() => {
    workspaceQuotaService.quota.revalidate();
  }, [workspaceQuotaService]);
```